### PR TITLE
Get record from correct lazy entry view. HD and other memory formats …

### PR DESCRIPTION
**Pull request by ahmetmircik**
…have different lazy views

ee counterpart https://github.com/hazelcast/hazelcast-enterprise/pull/2779
Fixes: https://github.com/hazelcast/hazelcast-enterprise/issues/2780

- [ ] needs backport to 3.11

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/EvictBatchBackupOperation.java [hunk 1]
```diff
@@ -97,24 +93,7 @@ private void equalizeEntryCountWithPrimary() {
             return;
         }
 
-        Evictor evictor = mapContainer.getEvictor();
-        if (evictor != Evictor.NULL_EVICTOR) {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/EvictBatchBackupOperation.java]: This branch is now missing. What is the reason for this change?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/operation/EvictBatchBackupOperation.java]: reason is to align with jcache counterpart: https://github.com/hazelcast/hazelcast/blob/8b93dae4f5a27605a67b640649283503f7a1bd3b/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheExpireBatchBackupOperation.java#L65

+ decreasing complexity

Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/map/impl/operation/EvictBatchBackupOperation.java]: Is this a behaviour change? Meaning, previously if a map eviction policy was configured, the backup op would evict according to this policy while now it evicts randomly? Is this acceptable?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/operation/EvictBatchBackupOperation.java]: I see it as a simplification. That part of code contains the fallback case. If we are there, it means, backup has different entries than primary, in this case, there is no fully correct way to find removal candidates + removing entries via eviction does lots of sampling for each entry compared to direct sampling, and this increases latency + https://github.com/hazelcast/hazelcast/pull/14609#discussion_r261574656
